src/EditorFeatures/CSharpTest/Completion/CompletionProviders/OverrideCompletionProviderTests.cs

```diff
@@ -3083,5 +3096,40 @@ public async Task NoCloneInOverriddenRecord()
     override $$
 }", cloneMemberName);
         }
+
+        [WpfFact, Trait(Traits.Feature, Traits.Features.Completion)]
+        [WorkItem(48640, "https://github.com/dotnet/roslyn/issues/48640")]
+        public async Task ObjectEqualsInClass()
+        {
+            await VerifyItemIsAbsentAsync(@"
+class Program 
```

Comment:
If overriding `bool Equals(object obj)` is invalid in the class case, should we indicate this to the user like we do with records (i.e. via red squiggle)?

Reply:
Equals(object) is fine in the normal case.  It's only invalid in the record case.  But we detect htat just by looking for an implicit overridde of that member as only records do that.